### ISSUE TITLE
[Feature] Support Bazel in `WasmEdge-go`

### ISSUE THREAD

**Issue by DarumaDocker** [WasmEdge/WasmEdge#2162]
[Wasmtime provides the compling support with Bazel](https://github.com/bytecodealliance/wasmtime-go)

> This Go library uses CGO to consume the C API of the [Wasmtime project](https://github.com/bytecodealliance/wasmtime) which is written in Rust. Precompiled binaries of Wasmtime are checked into this repository on tagged releases so you won't have to install Wasmtime locally, but it means that this project only works on Linux x86_64, macOS x86_64 , and Windows x86_64 currently. Building on other platforms will need to arrange to build Wasmtime and use CGO_* env vars to compile correctly.
>
> This project has been tested with Go 1.13 or later.
>
> If you are a bazel user, add following to your WORKSPACE file:
>
> ```
> go_repository(
>    name = "com_github_bytecodealliance_wasmtime_go",
>    importpath = "github.com/bytecodealliance/wasmtime-go/v3",
>    version = "v3.0.2",
> )
> ```

If Bazel is very popular in Golang community, should we also support it in WasmEdge-go?

**Comment by arnabnandikgp** [comment]
@hydai  is anyone working on this issue as I am interested in working on it.

**Comment by hydai** [comment]
Hi @arnabnandikgp 
Nope. It's all yours now :-)

**Comment by arnabnandikgp** [comment]
I have added the necessary WORKSPACE and BUILD file in my local setup but am not able to build any example specified in the WasmEdge-go-Examples repository using bazel and getting the 'missing input file' error from bazel though I have verified whether the location of src files being sourced from wasmedge-go/wasmedge is correct but am still getting this error. Moreover in the cc_library rule in the bazel BUILD file we need to add the c++ static library of wasmedge to compile so can anyone tell me which static library i should use for this purpose as there are multiple.

**Comment by hydai** [comment]
Hi @arnabnandikgp 
Could you please share the reproduction steps and logs you got?
And @q82419 may help with this issue.

**Comment by kalamkaar9404** [comment]
Hi @hydai, I would like to work on this issue. Can you please assign this to me?

**Comment by hydai** [comment]
Hi @kalamkaar9404 
Assigned to you.

**Comment by hydai** [comment]
Hi @kalamkaar9404,  
Do you have any updates? Or is it okay to open this to other people for this feature?

**Comment by kalamkaar9404** [comment]
Hello @hydai,

I’ve successfully completed the build process and am currently working on resolving the remaining dependency issues to ensure all tests run correctly. I expect to have all tests passing within the next few days and will submit it for review shortly after.

Thank you!

**Comment by hydai** [comment]
Hi @kalamkaar9404 
Are you still working on this issue?

**Comment by kalamkaar9404** [comment]
Yes @hydai. Apologies for the delay but I am still working on the issue to resolve the errors that I am getting while running the tests having completed the build successfully.

**Comment by kalamkaar9404** [comment]
I would be grateful if you could help me or suggest me how to move forward.

**Comment by kalamkaar9404** [comment]
Hello @hydai 
I had to pause work on this issue a few months ago due to severe health issues, but I’m doing better now and have resumed working on it. I’ll share updates soon. 
Thanks for your patience!

**Comment by kalamkaar9404** [comment]
Hi @hydai ! I have submitted a Pull Request to the WasmEdge-go repository to address the Bazel support for Windows.

You can find the PR here: [second-state/WasmEdge-go#58]

I have verified the changes on my fork, and the CI build is successfully passing for all platforms. Please let me know if any further changes are needed! Thanks
